bisection.go

```diff
@@ -7,11 +7,14 @@ package optimize
 import "math"
 
 // Bisection is a Linesearcher that uses a bisection to find a point that
-// satisfies the strong Wolfe conditions with the given gradient constant and
-// function constant of zero. If GradConst is zero, it will be set to a reasonable
-// value. Bisection will panic if GradConst is not between zero and one.
+// satisfies the strong Wolfe conditions with the given curvature factor and
+// sufficient decrease factor of zero.
```

Comment:
Yea, there should probably be a DecreaseFactor. The internal logic may need to change some though, so if you want to do that in a different PR that's fine.